### ISSUE TITLE
Support all PEP 440 version specifiers

### ISSUE THREAD

**Issue by rpanderson** [chrisjbillington/setuptools-conda#8]
[PEP 440 version specifiers](https://www.python.org/dev/peps/pep-0440/#version-specifiers) such as `~=` are not supported.

**Comment by chrisjbillington** [comment]
Most are, just not `~=` since it doesn't have a simple conda equivalent.

Can probably translate it into two requirements: less than next major version, greater than or equal to specified version. Will read the spec to confirm.

**Comment by rpanderson** [comment]
Parsing using [`packaging.specifiers`](https://packaging.pypa.io/en/latest/specifiers.html) might suffice.

**Comment by chrisjbillington** [comment]
Addressed in 7ba435652594aa35dd4bb41f4eaa2695a9d92049.

Could you test?

packaging.specifiers didn't really help other than using its code as a reference, so it was a bit tricky. But refactoring the setuptools-conda code that converts version requirements into separate bits for package names, version specifiers, and environment markers was a good cleanup to have done anyway.

**Comment by rpanderson** [comment]
Thanks, @chrisjbillington. This 'worked' but the  `~=` version specifier was parsed for `conda` as, e.g. `numpy >=1.19.5,==1.19.*` resulting in
```
WARNING:conda.models.version:Using .* with relational operator is superfluous and deprecated and will be removed in a future version of conda. Your spec was 1.19.*, but conda is ignoring the .* and treating it as 1.19
```
Should this instead be `numpy >=1.19.5,<2`? 

See:
https://docs.conda.io/projects/conda-build/en/latest/resources/package-spec.html#package-match-specifications

**Comment by chrisjbillington** [comment]
`numpy >=1.19.5,<2` would work, but would require to split and do arithmetic. Nontrivial given there can be all sorts of bits in there like "dev" "pre" "post", "rc" and whatnot.

If I'm reading right, conda `==1.19` is equivalent to PEP440 `==1.19.*`. I'll just remove any `.*` from the end of specifiers (and not add it, in the case of the equivalent specification for the `~=` operator).

**Comment by chrisjbillington** [comment]
Should be resolved in 889c4093613a9885d7f0508137d8db76c05e3cce

`numpy ~=1.19.5` should now translate as `numpy >=1.19.5,==1.19`

And also `numpy ==1.19.*` should now translate as `numpy ==1.19`

**Comment by rpanderson** [comment]
LGTM.
